inference-engine/thirdparty/clDNN/kernel_selector/core/cl_kernels/gemm_mmad_int8.cl

```diff
@@ -0,0 +1,544 @@
+#define ACTIVATION_TYPE_VEC     CAT(ACTIVATION_TYPE, SUB_GROUP_SIZE)
+#define PACKED_INPUT0_TYPE_VEC  CAT(PACKED_INPUT0_TYPE, SUB_GROUP_SIZE)
+#define PACKED_INPUT1_TYPE_VEC  CAT(PACKED_INPUT1_TYPE, SUB_GROUP_SIZE)
+#define BLOCK_READ(ptr)         intel_sub_group_block_read((const __global uint*)(ptr))
+#define BLOCK_SHUFFLE           intel_sub_group_shuffle
+
+#if SUB_GROUP_SIZE == 8
+#define MMAD                    MMAD_8x8
+#else // SUB_GROUP_SIZE == 8
+#define MMAD                    MMAD_16x16
+#define TILE_SIZE_M_DIV         (TILE_SIZE_M / 2)
+#endif // SUB_GROUP_SIZE == 8
+
+inline uint FUNC(get_input0_batch_offset)(uint b, uint f, uint w, uint z) {
+#if INPUT0_SIMPLE
+    return GET_DATA_INDEX_6D_SAFE(INPUT0, b, f, w, z, 0, 0);
+#else // INPUT0_SIMPLE
+#   error gemm_mmad_int8.cl : Unsupported input 0 format
+#endif // INPUT0_SIMPLE
+}
+
+inline uint FUNC(get_input1_batch_offset)(uint b, uint f, uint w, uint z) {
+#if INPUT1_SIMPLE
+    return GET_DATA_INDEX_6D_SAFE(INPUT1, b, f, w, z, 0, 0);
+#else // INPUT1_SIMPLE
+#   error gemm_mmad_int8.cl : Unsupported input 1 format
+#endif // INPUT1_SIMPLE
+}
+
+#ifdef INPUT2_TYPE
+inline uint FUNC(get_input2_batch_offset)(uint b, uint f, uint w, uint z) {
+#if INPUT2_SIMPLE
+    return GET_DATA_INDEX_6D_SAFE(INPUT2, b, f, w, z, 0, 0);
+#else // INPUT2_SIMPLE
+#   error gemm_mmad_int8.cl : Unsupported input 2 format
+#endif // INPUT2_SIMPLE
+}
+#endif // INPUT2_TYPE
+
+inline uint FUNC(get_output_batch_offset)(uint b, uint f, uint w, uint z) {
+#if OUTPUT_SIMPLE
+    return GET_DATA_INDEX_6D(OUTPUT, b, f, w, z, 0, 0);
+#else // OUTPUT_SIMPLE
+#   error gemm_mmad_int8.cl : Unsupported output format
+#endif // OUTPUT_SIMPLE
+}
+
+__attribute__((reqd_work_group_size(SUB_GROUP_SIZE, 1, 1)))
+__attribute__((intel_reqd_sub_group_size(SUB_GROUP_SIZE)))
+KERNEL(gemm_mmad_int8)(
+    const __global INPUT0_TYPE* input0,
+    const __global INPUT1_TYPE* input1,
+#ifdef INPUT2_TYPE
+    const __global INPUT2_TYPE* input2,
+#endif // INPUT2_TYPE
+    __global OUTPUT_TYPE* output
+#if HAS_FUSED_OPS_DECLS
+    , FUSED_OPS_DECLS
+#endif // HAS_FUSED_OPS_DECLS
+    )
+
+// ***************************************************************************************** //
+// Kernel with leftovers for all sizes of input matrices and all transposition combinations. //
+// ***************************************************************************************** //
+
+#if OUTPUT_LEFTOVERS_M || OUTPUT_LEFTOVERS_N || OUTPUT_LEFTOVERS_K
+{
+    const uint output_x = (uint)get_global_id(0);
+    const uint output_x_tile = output_x / TILE_SIZE_N;
+    const uint output_y_tile = (uint)get_global_id(1);
+    uint batch = get_global_id(2);
+    const uint lid = (uint)get_local_id(0);
+
+    const uint b = batch % OUTPUT_BATCH_NUM;
+    batch /= OUTPUT_BATCH_NUM;
+    const uint f = batch % OUTPUT_FEATURE_NUM;
+    batch /= OUTPUT_SIZE_Z;
+    const uint w = batch % OUTPUT_SIZE_W;
+    batch /= OUTPUT_FEATURE_NUM;
```

Comment:
Are you sure that these indices are calculated correctly? 
Looks like it should be something like:
```c
    const uint f = batch % OUTPUT_FEATURE_NUM;
    batch /= OUTPUT_FEATURE_NUM;
    const uint w = batch % OUTPUT_SIZE_W;
    batch /= OUTPUT_SIZE_W;
```